lib/resty/etcd/v3.lua

```diff
@@ -31,7 +31,34 @@ local mt = { __index = _M }
 -- define local refresh function variable
 local refresh_jwt_token
 
-local function _request_uri(self, endpoint, method, uri, opts, timeout, ignore_auth)
+
+local function choose_endpoint(self)
+    local endpoints = self.endpoints
+    local endpoints_len = #endpoints
+    if endpoints_len == 1 then
+        return endpoints[1]
+    end
+
+    if health_check.conf ~= nil then
+        for _, endpoint in ipairs(endpoints) do
+            if health_check.get_target_status(endpoint.http_host) then
+                return endpoint
+            end
+        end
+        return nil, "has no healthy etcd endpoint available"
+    end
+
+    self.init_count = (self.init_count or 0) + 1
+    local pos = self.init_count % endpoints_len + 1
```

Comment:
Is it intended to skip the first endpoint? As the minimal init_count here is 1.

Reply:
I also failed to figure out the purpose of this. It's the implementation of [the earliest v3 API](https://github.com/api7/lua-resty-etcd/blob/44983b54cbc9a5c4b7cd51df5b07037ba22e60b6/lib/resty/etcd/v3.lua#L141-L142).
Hi @nic-chen do you remember why is it set so

Reply:
how about `self.init_count = (self.init_count or -1) + 1` ?

Reply:
I also don’t remember, it’s contributed by the community

Reply:
Got it. Fixed as suggested. Thanks!